### ISSUE TITLE
strerror_r returns int in emscripten

### ISSUE THREAD

**Issue by OFFTKP** [ToruNiina/toml11#235]
exception.hpp:44 fails to compile with error `error: cannot initialize a variable of type 'const char *' with an rvalue of type 'int'` using emscripten

**Comment by ToruNiina** [comment]
I removed `strerror` that causes this problem because of its high maintenance-cost. So it should be fixed with the latest commit.
I will soon release patch v3.8.1 after CI passes (maybe a few hours later).

----

Background:

`strerror` is used only when you pass `FILE*` to `toml::parse` instead of `std::string filename`, `std::filesystem::path`, or `std::istream`.

There are multiple variants of `strerror` depending on the environment. The standard version is not thread-safe, so some compilers complain about it (already reported). To avoid this problem, we use environment-specific macros to detect which one is available. But I found that there are multiple environments/conditions where we cannot detect the available variant of `strerror` using de-facto macro (actually, we have just fixed the same kind of problem yesterday, before the release). There can be more envs where this does not work. The benefit of using `strerror` is not worth the cost.

I think the `FILE*` version is less used than other overloads (most of the users may not even know such an overload exists). And the impact is even smaller, since the change does not remove the ability to pass `FILE*`, but only to return an error number instead of a string if it fails. Still, one can search for the error number, or can get the errno from the exception and call the variant of `strerror` that is appropriate in the environment.
